horizons/world/player.py

```diff
@@ -69,7 +69,8 @@ def initialize(self, inventory):
 
 	def __init(self, name, color, clientid, difficulty_level, max_tier_notification, settlerlevel=0):
 		assert isinstance(color, Color)
-		assert isinstance(name, basestring) and name
+		assert isinstance(name, basestring)
+		assert name
```

Comment:
Type hints aren't enforced by the interpreter AFAIK. Third-party libraries may use that to provide some kind of static type checking.

I see it as a way of documentation, and possibly later, some static type checks on the codebase (not during runtime).